paddle/fluid/imperative/layer.cc

```diff
@@ -219,6 +222,23 @@ void VarBase::ClearGradient() {
   }
 }
 
+void VarBase::_GradientSetEmpty(bool set_is_empty) {
+  VLOG(4) << "Gradient " << Name() << " SetIsEmpty " << set_is_empty;
+  if (grad_var_) {
+    grad_var_->SharedVar()->SetIsEmpty(set_is_empty);
+  }
+}
+
+bool VarBase::_IsGradientSetEmpty() {
+  bool res = true;
+  if (grad_var_) {
+    res = grad_var_->SharedVar()->is_empty_;
```

Comment:
not safe an access of SharedVar, check if it's nullptr first

Reply:
Done.